recipes/mysql-connector-cpp/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake
+from conan.tools.files import get
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd, cross_building, stdcpp_library
+
+class MysqlConnectorCPPRecipe(ConanFile):
+    name = "mysql-connector-cpp"
+    license = "GPL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://dev.mysql.com/doc/connector-cpp/8.0"
+    description = "A Conan package for MySQL Connector/C++ with OpenSSL, Boost, and libmysqlclient"
+    topics = ("mysql", "connector", "cpp", "openssl", "boost", "libmysqlclient", "jdbc", "static")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "9" if Version(self.version) >= "8.0.27" else "5.3",
+            "clang": "6",
+        }
+
+    def requirements(self):
+        self.requires("boost/1.81.0")
+        # build faiils on mac with openssl 3x
+        if self.settings.os == "Macos":
+            self.requires("openssl/1.1.1t")
+        else:
+            self.requires("openssl/[>=1.1 <4]")
+        self.requires("libmysqlclient/8.0.31")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self.source_folder)
+
+    def validate_build(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        if hasattr(self, "settings_build") and cross_building(self, skip_x64_x86=True):
+            raise ConanInvalidConfiguration("Cross compilation not yet supported by the recipe. Contributions are welcomed.")
+
+    def validate(self):
```

Comment:
```suggestion
    def validate_build(self):
        if hasattr(self, "settings_build") and cross_building(self, skip_x64_x86=True):
            raise ConanInvalidConfiguration("Cross compilation not yet supported by the recipe. Contributions are welcomed.")

    def validate(self):
        if self.settings.compiler.get_safe("cppstd"):
            check_min_cppstd(self, self._min_cppstd)

```
checking cppstd is always in validate because Conan client should consider which cppstd is supported when installing a package too.